_posts/2018-11-26-Square-CTF-2018-Gate-of-hell.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Square CTF 2018 - Gates of hells WriteUp
+layout: post
+author: "XxcoralloxX"
+tags: reverse TuCTF
+category: reverse
+---
+
+This is a reverse challenge.
+As we will see, this is probably a code written directly in assembly, and not some C code compilated.
+Since the code is quite short, we will be able to read it all.
+
+First, we try running the binary:
+
+![AltText](https://i.gyazo.com/4918234b46d98a0e21bf0eb77040e3a0.png)
+
+Nothing happens.
+
+Have a look inside:
+
+![AltText](https://i.gyazo.com/6e53f9bd4585f186ec78d2cd279eaab6.png)
+
+We can see that during the first 3 instruction 
+a pop takes the number of arguments and compares it with 0x10 (16).
+
+![AltText](https://i.gyazo.com/3288115b512b7441dbaa69caa3ac9332.png)
+
+So we need to provide 15 arguments? (the first is the name of the program).
+
+Ok, 15 arguments, but when will we win?
+we can find this function
+
+![AltText](https://i.gyazo.com/93807576fe368fb6a98601cfc369db6f.png)
+
+which is a system call to a print. (which is probably our win function).
+
+as we can see this function will be called if Ebx will be equal at 0x29A (666)
```

Comment:
One thing that could be useful for the next writeups is enclosing register names in a rich text format, like `ebx`, _ebx_ or **ebx**